src/core/public/application/application_service.test.ts

```diff
@@ -676,6 +708,53 @@ describe('#start()', () => {
       expect(MockHistory.push).toHaveBeenCalledWith('/custom/path#/hash/router/path', undefined);
     });
 
+    it('preserves trailing slash when path contains a hash', async () => {
+      const { register } = service.setup(setupDeps);
+
+      register(Symbol(), createApp({ id: 'app2', appRoute: '/custom/path' }));
```

Comment:
is it used in the test?

Reply:
Adapted this rather quickly. The `navigateToApp('myTestApp', { path: '#/' })` below should be using this `app2` instead. Will fix.